### PR TITLE
feat: optimize NFT avatar's price display rule

### LINKED ISSUE #5009: [Demand] Optimize NFT avatar's price display rule
## Catalogue

- [ ] UI/UX
- [ ] Mask Core Feature
- [x] Mask Plugin Feature
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background
Currently, we set NFT avatar display info from fetched data, but some NFT doesn't have price and some NFT collections are very special.  

## Part 2: Problem to Solve
Optimize the price display rule.

## Part 3: Expected Behavior
Common rule:
1. If the price is 0, don't need to display price info.

Special rule:
1. If the NFT is ENS collection, append text "ENS" in the end.

image

### PR DESCRIPTION
**Pull request by lelenei**
closes #5009

### CODE REVIEW

File: packages/mask/src/plugins/Avatar/SNSAdaptor/NFTBadge.tsx [hunk 1]
```diff
@@ -40,13 +40,14 @@ interface NFTBadgeProps extends withClasses<'root' | 'text' | 'icon'> {
     width?: number
 }
 
-function formatPrice(amount: string) {
+function formatPrice(amount: string, symbol: string, slug: string) {
```
Comment (zhouhanseng) [inline packages/mask/src/plugins/Avatar/SNSAdaptor/NFTBadge.tsx:43]: No need to pass `slug` and `symbol` into `formatPrice()`. Just handle them in `formatText()`

File: packages/mask/src/plugins/Avatar/SNSAdaptor/NFTBadge.tsx [hunk 2]
```diff
@@ -62,8 +62,8 @@ export function NFTBadge(props: NFTBadgeProps) {
                     strokeWidth={14}
                     stroke="black"
                     fontSize={9}
-                    text={loading ? 'loading...' : name}
-                    price={loading ? '' : `${formatPrice(amount)} ${symbol}`}
+                    text={loading ? 'loading...' : `${name} ${slug.toLocaleLowerCase() === 'ens' ? 'ENS' : ''}`}
```
Comment (septs) [inline packages/mask/src/plugins/Avatar/SNSAdaptor/NFTBadge.tsx]:
```suggestion
                    text={loading ? 'loading...' : `${name} ${slug.toLowerCase() === 'ens' ? 'ENS' : ''}`}
```

**Comment by orange187** [comment]
Checked

File: packages/mask/src/plugins/Avatar/SNSAdaptor/NFTBadge.tsx [hunk 3]
```diff
@@ -33,8 +33,12 @@ function formatPrice(amount: string, symbol: string) {
...
 }
 
-function formatText(symbol: string, length: number) {
-    return symbol.length > length ? `${symbol.slice(0, length)}...` : symbol
+function formatText(name: string, tokenId: string) {
+    const _name = name.replace(/#\d*/, '')
+    if (tokenId.length > 10) {
+        return `${_name.trim()} #${tokenId.substr(0, 6)}...${tokenId.substr(-4)}`
+    }
+    return `${_name} #${tokenId}`
```
Comment (septs) [inline packages/mask/src/plugins/Avatar/SNSAdaptor/NFTBadge.tsx]:
```suggestion
    name = name.replace(/#\d*/, '').trim()
    tokenId = wrap(token)
    if (tokenId.length > 10) {
        token = tokenId.slice(0, 6) + '...' + tokenId.slice(-4)
    }
    return `${name} #${tokenId}`
```

plz DO NOT USE `substr` and `strstring`, thanks

**Review by Jack-Works — DISMISSED** [review]

**Comment by wenluomask** [comment]
Checked.

File: packages/web3-providers/src/opensea/types.ts [hunk 4]
```diff
@@ -113,15 +113,15 @@ export interface OpenSeaResponse extends Asset {
...
-interface AssetEvent {
-    event_type: string
-    event_timestamp: number
-    auction_type: string
-    total_price: string
-    transaction: Transaction | null
-    payment_token: OpenSeaFungibleToken | null
-}
+//
+// interface AssetEvent {
```
Comment (guanbinrui) [inline packages/web3-providers/src/opensea/types.ts]: Remove commented code.

**Pull request merged**